modules/porous_flow/src/kernels/PorousFlowHeatVolumetricExpansion.C

```diff
@@ -0,0 +1,104 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "PorousFlowHeatVolumetricExpansion.h"
+
+template<>
+InputParameters validParams<PorousFlowHeatVolumetricExpansion>()
+{
+  InputParameters params = validParams<TimeKernel>();
+  params.addRequiredParam<UserObjectName>("PorousFlowDictator", "The UserObject that holds the list of Porous-Flow variable names.");
+  params.addClassDescription("Energy-density*rate_of_solid_volumetric_expansion");
+  return params;
+}
+
+PorousFlowHeatVolumetricExpansion::PorousFlowHeatVolumetricExpansion(const InputParameters & parameters) :
+    TimeKernel(parameters),
+    _dictator(getUserObject<PorousFlowDictator>("PorousFlowDictator")),
+    _var_is_porflow_var(_dictator.isPorousFlowVariable(_var.number())),
+    _num_phases(_dictator.numPhases()),
+    _fluid_present(_num_phases > 0),
+    _porosity(getMaterialProperty<Real>("PorousFlow_porosity_nodal")),
+    _dporosity_dvar(getMaterialProperty<std::vector<Real> >("dPorousFlow_porosity_nodal_dvar")),
+    _dporosity_dgradvar(getMaterialProperty<std::vector<RealGradient> >("dPorousFlow_porosity_nodal_dgradvar")),
+    _rock_energy_nodal(getMaterialProperty<Real>("PorousFlow_matrix_internal_energy_nodal")),
+    _drock_energy_nodal_dvar(getMaterialProperty<std::vector<Real> >("dPorousFlow_matrix_internal_energy_nodal_dvar")),
+    _fluid_density(_fluid_present ? &getMaterialProperty<std::vector<Real> >("PorousFlow_fluid_phase_density") : NULL),
```

Comment:
You're going to start wanting to use `nullptr` instead of `NULL`. The reason is typing. The former is actually a pointer type while the later is an unsigned int. The later can fail in newer C++11 code in `auto` declarations and in `initializer lists`. It's fine here but moving to the new keyword is always preferred.